### PR TITLE
fix wallet not working issue

### PR DESCRIPTION
**Pull request by chaosma**
code cleaning. remove LeaderMap in txgen and wallet

### CODE REVIEW

File: cmd/client/wallet/main.go [hunk 1]
```diff
@@ -17,10 +17,12 @@ import (
 	crypto2 "github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/params"
-	clientService "github.com/harmony-one/harmony/api/client/service"
+
+	//clientService "github.com/harmony-one/harmony/api/client/service"
```
Comment (LeoHChen) [inline cmd/client/wallet/main.go]: ? just remove it if not needed.

File: cmd/client/wallet/main.go [hunk 2]
```diff
@@ -308,28 +310,37 @@ func convertBalanceIntoReadableFormat(balance *big.Int) string {
...
+	peers := []p2p.Peer{}
+	walletNode.BeaconNeighbors.Range(func(k, v interface{}) bool {
+		peers = append(peers, v.(p2p.Peer))
+		return true
+	})
+
+	for peer := range peers {
+		fmt.Printf("[test] peer is: %v\n", peer)
 	}
+	//	for shardID, leader := range walletNode.Client.Leaders {
```
Comment (LeoHChen) [inline cmd/client/wallet/main.go]: just remove it if not needed. Just add some TODO comments.

File: cmd/client/wallet/main.go [hunk 3]
```diff
@@ -308,28 +310,37 @@ func convertBalanceIntoReadableFormat(balance *big.Int) string {
...
-	for shardID, leader := range walletNode.Client.Leaders {
-		port, _ := strconv.Atoi(leader.Port)
-		client := clientService.NewClient(leader.IP, strconv.Itoa(port+node.ClientServicePortDiff))
-		response := client.GetFreeToken(address)
-
-		txID := common.Hash{}
-		txID.SetBytes(response.TxId)
-		fmt.Printf("Transaction Id requesting free token in shard %d: %s\n", int(shardID), txID.Hex())
-	}
+	//	for shardID, leader := range walletNode.Client.Leaders {
```
Comment (LeoHChen) [inline cmd/client/wallet/main.go]: just remove it and leave some TODO comment here should be fine.

File: cmd/client/wallet/main.go (hunk 1 above)
Comment (chaosma) [reply cmd/client/wallet/main.go]: will added in a new pull request later

File: cmd/client/wallet/main.go (hunk 2 above)
Comment (chaosma) [reply cmd/client/wallet/main.go]: will removed

File: cmd/client/wallet/main.go (hunk 3 above)
Comment (chaosma) [reply cmd/client/wallet/main.go]: will removed

File: api/service/config.go [hunk 4]
```diff
@@ -13,11 +13,14 @@ import (
 // cyclic imports
 type NodeConfig struct {
 	// The three groupID design, please refer to https://github.com/harmony-one/harmony/blob/master/node/node.md#libp2p-integration
-	Beacon   p2p.GroupID                    // the beacon group ID
-	Group    p2p.GroupID                    // the group ID of the shard
-	Client   p2p.GroupID                    // the client group ID of the shard
+	Beacon p2p.GroupID // the beacon group ID
```
[the remark this answers is not in the document]
Comment (chaosma) [reply api/service/config.go]: caused by the comment line below

File: api/service/discovery/service.go [hunk 5]
```diff
@@ -72,7 +72,30 @@ func (s *Service) Run() {
...
+			}
+		case <-s.stopChan:
+			utils.GetLogInstance().Debug("wallet: stop discovery")
+			return
+		}
+	}
+}
+
 func (s *Service) contactP2pPeers() {
+	if s.config.IsWallet {
```
[the remark this answers is not in the document]
Comment (chaosma) [reply api/service/discovery/service.go]: moved to the Run function

File: api/service/networkinfo/service.go [hunk 6]
```diff
@@ -118,7 +118,7 @@ func (s *Service) DoService() {
 		select {
 		case peer := <-s.peerInfo:
 			if peer.ID != s.Host.GetP2PHost().ID() && len(peer.ID) > 0 {
-				//				utils.GetLogInstance().Info("Found Peer", "peer", peer.ID, "addr", peer.Addrs, "my ID", s.Host.GetP2PHost().ID())
+				utils.GetLogInstance().Info("Found Peer", "peer", peer.ID, "addr", peer.Addrs, "my ID", s.Host.GetP2PHost().ID())
```
[the remark this answers is not in the document]
Comment (chaosma) [reply api/service/networkinfo/service.go]: comment it out

File: api/service/discovery/service.go [hunk 7]
```diff
@@ -72,7 +72,30 @@ func (s *Service) Run() {
 	go s.contactP2pPeers()
 }
 
+func (s *Service) addP2pPeers() {
+	for {
+		select {
+		case peer, ok := <-s.peerChan:
+			utils.GetLogInstance().Debug("addP2pPeers", "peer", peer)
```
[the remark this answers is not in the document]
Comment (chaosma) [reply api/service/discovery/service.go]: removed

File: node/service_setup.go [hunk 8]
```diff
@@ -110,6 +122,8 @@ func (node *Node) ServiceManagerSetup() {
 		node.setupForNewNode()
 	case nodeconfig.ClientNode:
 		node.setupForClientNode()
+	case nodeconfig.WalletNode:
```
Comment (mikedoan) [inline node/service_setup.go]: why WalletNode - thought after offline chat, we would go to unify ClientNode logic

**Pull request merged**